core/src/main/scala/org/apache/spark/h2o/converters/H2ORDD.scala

```diff
@@ -96,17 +96,15 @@ class H2ORDD[A <: Product: TypeTag: ClassTag, T <: Frame] private(@transient val
         val data = new Array[Option[Any]](numCols)
         // FIXME: this is not perfect since ncols does not need to match number of names
         (0 until numCols).foreach{ idx =>
-          val value = if (converterCtx.isNA(idx)) None
-          else types(idx) match {
-            case q if q == classOf[Integer]           => Some(converterCtx.getInt(idx))
-            case q if q == classOf[java.lang.Long]    => Some(converterCtx.getLong(idx))
-            case q if q == classOf[java.lang.Double]  => Some(converterCtx.getDouble(idx))
-            case q if q == classOf[java.lang.Float]   => Some(converterCtx.getFloat(idx))
-            case q if q == classOf[java.lang.Boolean] => Some(converterCtx.getBoolean(idx))
+          data(idx) = types(idx) match {
+            case q if q == classOf[Integer]           => Option(converterCtx.getInt(idx))
+            case q if q == classOf[java.lang.Long]    => Option(converterCtx.getLong(idx))
+            case q if q == classOf[java.lang.Double]  => Option(converterCtx.getDouble(idx))
+            case q if q == classOf[java.lang.Float]   => Option(converterCtx.getFloat(idx))
+            case q if q == classOf[java.lang.Boolean] => Option(converterCtx.getBoolean(idx))
             case q if q == classOf[String] => Option(converterCtx.getString(idx))
```

Comment:
Good point... what is the byte equal to before it's being assigned?

Thanks,
-Vlad

On Fri, Aug 26, 2016 at 10:19 AM, Jakub Háva <notifications@github.com>
wrote:

> In core/src/main/scala/org/apache/spark/h2o/converters/H2ORDD.scala
> <https://github.com/h2oai/sparkling-water/pull/60#discussion_r76454746>:
>
> > @@ -96,17 +96,15 @@ class H2ORDD[A <: Product: TypeTag: ClassTag, T <: Frame] private(@transient val
> >          val data = new Array[Option[Any]](numCols)
> >          // FIXME: this is not perfect since ncols does not need to match number of names
> >          (0 until numCols).foreach{ idx =>
> > -          val value = if (converterCtx.isNA(idx)) None
> > -          else types(idx) match {
> > -            case q if q == classOf[Integer]           => Some(converterCtx.getInt(idx))
> > -            case q if q == classOf[java.lang.Long]    => Some(converterCtx.getLong(idx))
> > -            case q if q == classOf[java.lang.Double]  => Some(converterCtx.getDouble(idx))
> > -            case q if q == classOf[java.lang.Float]   => Some(converterCtx.getFloat(idx))
> > -            case q if q == classOf[java.lang.Boolean] => Some(converterCtx.getBoolean(idx))
> > +          data(idx) = types(idx) match {
> > +            case q if q == classOf[Integer]           => Option(converterCtx.getInt(idx))
> > +            case q if q == classOf[java.lang.Long]    => Option(converterCtx.getLong(idx))
> > +            case q if q == classOf[java.lang.Double]  => Option(converterCtx.getDouble(idx))
> > +            case q if q == classOf[java.lang.Float]   => Option(converterCtx.getFloat(idx))
> > +            case q if q == classOf[java.lang.Boolean] => Option(converterCtx.getBoolean(idx))
> >              case q if q == classOf[String] => Option(converterCtx.getString(idx))
>
> If it would return Option, then all these calls
>
> mutableRow.setByte(idx, converterCtx.getByte(i)) would have to be like
> mutableRow.setByte(idx, converterCtx.getByte(i).getOrElse(null)) because
> we need to propagate null here
>
> In this case I would rather return like this and create Option out of it
> when needed, but it's probably just matter of style.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/h2oai/sparkling-water/pull/60/files/68c2c60f843cc57b33fffb7f988657423d3a3ee5#r76454746>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOuK-cZL95RZdPawDcYIg9hPwBCccNbks5qjyA2gaJpZM4Jt2EA>
> .
>